### DISCUSSION TITLE
How to change the primary key

### DISCUSSION THREAD

**Discussion by josias-r** [directus/directus#4011] [Archive]
I'd like to change the primary key from an existing collection. I don't think directus supports this in the UI currently unless I am blind. Maybe consider adding this feature of marking another field as the primary key (maybe combine with a "unique" flag for fields, that would be nice to have as well)?

Anyways, in my specific example, I'd like to switch from UUID to an auto-increment integer as the primary key. What steps do I need to take to change that? (I'm okay with losing the data, but an explanation on how to change the key without losing the data might be helpful for others)

Is editing the `direcuts_fields` and the collection table itself all it needs, or is there something else I need to adjust in the DB?

**Comment by rijkvanzanten** [comment]
Just changing the primary key can be pretty involved, especially seeing it'll be used in relationships (as foreign keys) too. There's currently no way to do it in Directus.

> Is editing the direcuts_fields and the collection table itself all it needs, or is there something else I need to adjust in the DB?

Assuming you don't have any relational data associated to this collection, that should be it

**Reply by philleepflorence** [reply]
@rijkvanzanten  - Do the Directus Activity and Revisions Collections require or default to the PK ID of a collection?

### Reference
[Primary and Foreign Keys](https://docs.directus.io/concepts/relationships/#primary-and-foreign-keys)

**Reply by rijkvanzanten** [reply]
Yeah those have the PK stored to know what the item was that was altered
